### ISSUE TITLE
Need consistent approach in using pre/post/deploy hooks with certonly and renew subcommands 

### ISSUE THREAD

**Issue by rakesh-p** [certbot/certbot#9869]
I have been asked to post my question here https://community.letsencrypt.org/t/reasoning-behind-hooks-directory-available-only-for-certficate-renewal/211182/

The usage is not similar when it comes to hooks for certonly and renew subcommands.

-  Just like 'renew' subcommand, it is possible to specify hooks via flags like --pre-hook, --post-hook for certonly but it is not documented (e.g. in 'certbot -h certonly'. )
-  It is possible to use the directories/folders (pre/post/deploy) with scripts with the renew subcommand but not with the certonly subcommand

It would be nice to:
1. update the documentation for certonly to show flags --pre-hook etc.
2. support folders (pre/post/deploy) for certonly as well

thanks

**Comment by nuxwin** [comment]
See also : https://github.com/certbot/certbot/issues/9978

By reading your issue, I now understand better the problem I'm encountering. Thank.

> I have been asked to post my question here https://community.letsencrypt.org/t/reasoning-behind-hooks-directory-available-only-for-certficate-renewal/211182/
> 
> The usage is not similar when it comes to hooks for certonly and renew subcommands.
> 
>     * Just like 'renew' subcommand, it is possible to specify hooks via flags like --pre-hook, --post-hook for certonly but it is not documented (e.g. in 'certbot -h certonly'. )
> 
>     * It is possible to use the directories/folders (pre/post/deploy) with scripts with the renew subcommand but not with the certonly subcommand
> 
> 
> It would be nice to:
> 
>     1. update the documentation for certonly to show flags --pre-hook etc.
> 
>     2. support folders (pre/post/deploy) for certonly as well
> 
> 
> thanks

**Comment by ohemorange** [comment]
I agree that our documentation could be better! https://github.com/certbot/certbot/pull/9439 and https://github.com/certbot/certbot/pull/5067 are attempts to improve the docs, both of which stalled for various reasons.

As for supporting folders for `certonly`, I'm looking into seeing if there's a reason it was done this way (only for `renew`).
